lib/assert.js

```diff
@@ -89,14 +89,25 @@ x.throws = function (fn, err, msg) {
 	}
 
 	try {
+		var errMsg;
 		if (typeof err === 'string') {
-			var errMsg = err;
+			errMsg = err;
 			err = function (err) {
 				return err.message === errMsg;
 			};
 		}
 
 		assert.throws(fn, err, msg);
+
+		if (errMsg) {
+			return Error(errMsg);
+		}
+
+		try {
+			fn();
```

Comment:
Good call. I didn't consider the implications of calling `fn` outside of the call to core-assert.